core/src/main/scala/org/apache/spark/memory/DefaultMemoryManager.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.memory
+
+import java.nio.ByteBuffer
+
+import com.gemstone.gemfire.internal.snappy.UMMMemoryTracker
+import com.gemstone.gemfire.internal.snappy.memory.MemoryManagerStats
+import io.snappydata.collection.ObjectLongHashMap
+
+import org.apache.spark.{Logging, SparkEnv}
+import org.apache.spark.storage.BlockId
+
+
+class DefaultMemoryManager extends StoreUnifiedManager with Logging {
+
+  private val memoryForObject: ObjectLongHashMap[(String, MemoryMode)]
+  = ObjectLongHashMap.withExpectedSize[(String, MemoryMode)](16)
+
+
+  private val managerId = "DefaultMemoryManager"
+
+  override def acquireStorageMemoryForObject(objectName: String,
+      blockId: BlockId,
+      numBytes: Long,
+      memoryMode: MemoryMode,
+      buffer: UMMMemoryTracker,
+      shouldEvict: Boolean): Boolean = {
+    logDebug(s"Acquiring DefaultManager memory for $objectName $numBytes")
+    if (SparkEnv.get ne null) {
+      val success = SparkEnv.get.memoryManager.acquireStorageMemory(blockId, numBytes, memoryMode)
+      memoryForObject.addTo(objectName -> memoryMode, numBytes)
```

Comment:
better to store SparkEnv.get volatile value in a val rather than calling twice (just for the case where executor is just being closed and this can throw NPE in second call)

Reply:
Ok. Will change.